unit/daq/daq.ino

```diff
@@ -1,216 +1,67 @@
+        pinString = pinString.substring(1);
+        pin = NAME2PIN[pinString.toInt()];
+        return analogRead(pin);
+    }
+    pin = pinString.toInt();
+    return digitalRead(pin);
+}
 
-class SystemWriter {
-  public:
-    static void Initialize(SystemState &state, Commands &commands) {
-      InitDataCom();
-    }
-    static void Update(SystemState &state, Commands &commands) {
-      UpdateDataCom(state, commands.printDataCom);
-      commands.printDataCom = false;
-    }
-  private:
-    SystemWriter() {
-    }
-    static void InitDataCom() {
-      // Initialize Serial Comunications
-      Serial.begin(4800);
-      Serial.setTimeout(0);
-    }
-    static void UpdateDataCom(SystemState &state, boolean printDataCom) {
-      if (printDataCom) {
-        String serDataCom = "";
-        serDataCom += "atmospheric_temp";
-        serDataCom += state.atmosTemp;
-        serDataCom += ',';
-        serDataCom += "reservoir_temp";
-        serDataCom += state.nutrientSolutionTemp;
-        serDataCom += ',';
-        serDataCom += "ph";
-        serDataCom += state.nutrientResevoirPH;
-        serDataCom += ',';
-        serDataCom += "electrical_conductivity";
-        serDataCom += state.nutrientResevoirEC;
-        serDataCom += ',';
-        serDataCom += "light_intensity";
-        serDataCom += state.lightLevel;
-        serDataCom += ',';
-        serDataCom += "air_flow";
-        serDataCom += state.airFlow;
-        serDataCom += ',';
-        serDataCom += "dissolved_oxygen";
-        serDataCom += state.dissolvedOxygen;
-        serDataCom += ',';
-        serDataCom += "soil_moisture";
-        serDataCom += state.mediumMoisture;
-        Serial.println(serDataCom);
-      }
-      else
-      {
-        Serial.println("No Data");
-      }
+void postRequest(String input) {
+    String pinString, valueString;
+    uint8_t pin, value;
+    while (true) {
+        pinString = input.substring(input.indexOf('(') + 1, input.indexOf(','));
+        valueString = input.substring(input.indexOf(',') + 1, input.indexOf(')'));
+        if (pinString.indexOf('A') >= 0) {
+            pinString = pinString.substring(1);
+            pin = NAME2PIN[pinString.toInt()];
+            value = valueString.toInt();
+            analogWrite(pin, value);
+        } else {
+            pin = pinString.toInt();
+            value = valueString.toInt();
+            digitalWrite(pin, value);
+        }
+        if (input.indexOf(')') + 2 <= input.length() - 1) {
+            input = input.substring(input.indexOf(')') + 2);
+        } else {
+            break;
+        }
     }
-};
-
-SystemState state = *(SystemState::GetInstance());
-Commands commands = *(Commands::GetInstance());
+}
 
 void setup() {
-  SystemReader::Initialize(state, commands);
-  SystemWriter::Initialize(state, commands);
+    // put your setup code here, to run once:
+    pinMode(A0, OUTPUT);
+    pinMode(A7, INPUT);
+    Serial.begin(9600);
 }
```

Comment:
I should be able to do all the pinMode stuff from the cpp side! I want to never have to touch this file. Also make 9600 a var at the top of the file for easy reference